src/overlays/actors/ovl_En_Bigslime/z_en_bigslime.c

```diff
@@ -48,289 +243,2911 @@ const ActorInit En_Bigslime_InitVars = {
+        dynamicVtx = &sBigslimeDynamicVtx[this->dynamicVtxState][i];
+
+        // vector dot product between each dynamicVtx and the unit normal vector describing player's thrown direction
+        dotXYZ = (dynamicVtx->n.ob[0] * unitVecX + dynamicVtx->n.ob[1] * M_SQRT1_2 + dynamicVtx->n.ob[2] * unitVecZ) *
+                 0.001f;
+        if (dotXYZ < 0.01f) {
+            this->vtxSurfacePerturbation[i] = 0.0f;
+        } else {
+            this->vtxSurfacePerturbation[i] = dotXYZ * 0.5f;
+        }
+    }
+
+    this->windupPunchTimer = 27;
+    Animation_PlayOnce(&this->skelAnime, &D_0600F048);
+    EnBigslime_GekkoSfxInsideBigslime(this, NA_SE_EN_FROG_PUNCH2);
+    this->actionFunc = EnBigslime_WindupThrowPlayer;
+}
+
+void EnBigslime_WindupThrowPlayer(EnBigslime* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    f32 scale;
+    f32 invWindupPunchTimer;
+    Vtx* dynamicVtx;
+    Vtx* staticVtx;
+    s32 i;
+    s32 j;
+
+    this->windupPunchTimer--;
+    EnBigslime_UpdateCameraGrabPlayer(this, globalCtx);
+    if (this->windupPunchTimer > 0) {
+        invWindupPunchTimer = 1.0f / this->windupPunchTimer;
+        scale = cos_rad(this->windupPunchTimer * (M_PI / 27)) + 1.0f;
+        player->actor.world.pos.y = this->actor.world.pos.y + (this->actor.scale.y * -500.0f);
+
+        // Linearly interpolate gekkoRot.y --> this->actor.world.rot.y
+        this->gekkoRot.y += (s16)((s16)(this->actor.world.rot.y - this->gekkoRot.y) * invWindupPunchTimer);
+        this->gekkoPosOffset.x = Math_SinS(this->gekkoRot.y) * -50.0f;
+        this->gekkoPosOffset.z = Math_CosS(this->gekkoRot.y) * -50.0f;
+    } else {
+        if (this->windupPunchTimer == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_B_SLIME_REVERSE);
+            EnBigslime_GekkoSfxInsideBigslime(this, NA_SE_EN_FROG_PUNCH1);
+        }
+
+        scale = 0.5f - cos_rad(-this->windupPunchTimer * (M_PI / 5)) * 0.5f;
+        if (this->windupPunchTimer == -5) {
+            if (player->stateFlags2 & 0x80) {
+                player->actor.parent = NULL;
+                player->unk_AE8 = 100;
+            }
+
+            player->actor.velocity.y = 0.0f;
+            func_800B8D50(globalCtx, &this->actor, 10.0f, this->actor.world.rot.y, 10.0f, 4);
+            EnBigslime_SetupSetDynamicVtxThrowPlayer(this, globalCtx);
+        }
+
+        player->actor.world.pos.x =
+            (Math_SinS(this->actor.world.rot.y) * (1060.5f * this->actor.scale.x) * scale) + this->actor.world.pos.x;
+        player->actor.world.pos.z =
+            (Math_CosS(this->actor.world.rot.y) * (1060.5f * this->actor.scale.z) * scale) + this->actor.world.pos.z;
+        player->actor.world.pos.y =
+            this->actor.world.pos.y + (this->actor.scale.y * -500.0f) + (1060.5f * this->actor.scale.y * scale);
+        invWindupPunchTimer = 1.0f / (this->windupPunchTimer + 6);
+        // this->gekkoPosOffset *= (1 - invWindupPunchTimer)
```

Comment:
```suggestion
```
Is this needed?

Reply:
This isn't needed no, I only put it there because I found it to be a more intuitive way to think of the formula below personally. Is it less clear?